packages/composition/dbaas/exoscale/kafka.yml

```diff
@@ -0,0 +1,106 @@
+classes:
+  - .common
+
+parameters:
+  pkg.appcat.composition.dbaas.exoscale:
+    secretNamespaces:
+      kafka: syn-provider-exoscale-secrets
+
+  appcat:
+    compositions:
+      "exoscale.kafka.appcat.vshn.io":
+        commonPatchSets:
+          annotations: {}
+          labels: {}
+        commonResources:
+          observeClaimNamespace: {}
+        spec:
+          compositeTypeRef:
+            apiVersion: appcat.vshn.io/v1
+            kind: XExoscaleKafka
+          writeConnectionSecretsToNamespace: ${crossplane:namespace}
+          resources:
+            - base:
+                apiVersion: exoscale.crossplane.io/v1
+                kind: Kafka
+                metadata: {} # patched
+                spec:
+                  forProvider:
```

Comment:
My idea was to not set it and delegate this to the provider/exoscale. If exoscale updates the default kafka version we don't really want to manually update this, do we?

Reply:
Actually I'm not sure anymore if we really want to hardcore the version though it would be a good idea to lock our composition on a specific version, what do you think?

Reply:
> If exoscale updates the default kafka version we don't really want to manually update this, do we?

yes that would be the strongest argument against adding the version. I guess showing the version in the status field should be enough? Though now I have a problem with how do we comunicate to the customer the version of the service he is about to create?

Reply:
Adding the version to the status is a good point. I'll do that.

As for how to communicate what version a customer will get, I guess that is generally an issue we can't really solve in the composition. We should probably solve this through documentation? We'll probably need end user documentation for all DBaaS anyway, right?

Reply:
Ah didn't know that only one version is valid.

But yeah already exposing it with a default still seems sensible. That way once new versions become available they can be used without any changes. We can document the circumstances via the description.

But on the other hand someone might enter $random_number and wonder why it's not working...

Reply:
> We should probably solve this through documentation? We'll probably need end user documentation for all DBaaS anyway, right?

Yes, exactly!

For me the end-user docs are two parts:
1. The composite
2. The actual webpage where we describe everything

The composite has the ability to already expose a lot of information via `kubectl explain` which ties neatly into the idea to be as k8s native as possible.

Reply:
I like the enum idea! Let's for now only allow `3.2` and empty string, that should prevent confusion and future prove the composite for future versions

Reply:
We might want to add this approach for the rest of DBaaS.

Reply:
Open a [PR](https://github.com/vshn/component-appcat/pull/50) for that.